cpu/sam0_common/periph/i2c.c

```diff
@@ -48,148 +50,110 @@
 #define SERCOM_I2CM_CTRLA_MODE_I2C_MASTER SERCOM_I2CM_CTRLA_MODE(5)
 #endif
 
-/* static function definitions */
-static void _i2c_poweron(SercomI2cm *sercom);
-static void _i2c_poweroff(SercomI2cm *sercom);
-
 static int _start(SercomI2cm *dev, uint8_t address, uint8_t rw_flag);
 static inline int _write(SercomI2cm *dev, const uint8_t *data, int length);
 static inline int _read(SercomI2cm *dev, uint8_t *data, int length);
 static inline void _stop(SercomI2cm *dev);
-static inline int _wait_for_response(SercomI2cm *dev, uint32_t max_timeout_counter);
+static inline int _wait_for_response(SercomI2cm *dev, uint32_t max_timeout_cnt);
 
 /**
  * @brief Array holding one pre-initialized mutex for each I2C device
  */
-static mutex_t locks[] = {
-#if I2C_0_EN
-    [I2C_0] = MUTEX_INIT,
-#endif
-#if I2C_1_EN
-    [I2C_1] = MUTEX_INIT,
-#endif
-#if I2C_2_EN
-    [I2C_2] = MUTEX_INIT
-#endif
-#if I2C_3_EN
-    [I2C_3] = MUTEX_INIT
-#endif
-};
+static mutex_t locks[I2C_NUMOF];
 
-int i2c_init_master(i2c_t dev, i2c_speed_t speed)
+/**
+ * @brief   Shortcut for accessing the used I2C SERCOM device
+ */
+static inline SercomI2cm *dev(i2c_t bus)
+{
+    return i2c_config[bus].dev;
+}
+
+int i2c_init_master(i2c_t bus, i2c_speed_t speed)
 {
-    SercomI2cm *I2CSercom = 0;
-    gpio_t pin_scl = 0;
-    gpio_t pin_sda = 0;
-    gpio_mux_t mux;
-    uint32_t clock_source_speed = 0;
-    uint8_t sercom_gclk_id = 0;
-    uint8_t sercom_gclk_id_slow = 0;
 
     uint32_t timeout_counter = 0;
     int32_t tmp_baud;
 
-    switch (dev) {
-#if I2C_0_EN
-        case I2C_0:
-            I2CSercom = &I2C_0_DEV;
-            pin_sda = I2C_0_SDA;
-            pin_scl = I2C_0_SCL;
-            mux = I2C_0_MUX;
-            clock_source_speed = CLOCK_CORECLOCK;
-            sercom_gclk_id = I2C_0_GCLK_ID;
-            sercom_gclk_id_slow = I2C_0_GCLK_ID_SLOW;
-            break;
-#endif
-        default:
-            DEBUG("I2C FALSE VALUE\n");
-            return -1;
-    }
-
     /* DISABLE I2C MASTER */
-    i2c_poweroff(dev);
+    i2c_poweroff(bus);
 
     /* Reset I2C */
-    I2CSercom->CTRLA.reg = SERCOM_I2CS_CTRLA_SWRST;
-    while (I2CSercom->SYNCBUSY.reg & SERCOM_I2CM_SYNCBUSY_MASK) {}
+    dev(bus)->CTRLA.reg = SERCOM_I2CM_CTRLA_SWRST;
```

Comment:
This was `i2cs` and now `i2cm`. Expected?

Reply:
Yes.
I2C Sercom can be either master or slave. I2CS bit resets all registers related to slave mode which is not our case (as our aim is to reset our master I2C here). This must be an unspotted mistake from previous driver.